image-builder/image_builder.sh

```diff
@@ -57,12 +57,30 @@ EOT
 exit "${error}"
 }
 
-while getopts "ho:s:" opt
+# Maximum allowed size in MB for root disk
+MAX_IMG_SIZE_MB=2048
+
+FS_TYPE=${FS_TYPE:-"ext4"}
+
+# Maximum no of attempts to create a root disk before giving up
+MAX_ATTEMPTS=5
+
+ATTEMPT_NUM=0
+
+while getopts "ho:s:f:" opt
 do
 	case "$opt" in
 		h)	usage ;;
 		o)	IMAGE="${OPTARG}" ;;
-		s)	IMG_SIZE="${OPTARG}" ;;
+		s)	IMG_SIZE=${OPTARG}
+                        if [ ${IMG_SIZE} -lt 0 ]; then
+                                die "Image size has to be greater than 0 MB."
+                        fi
+                        if [ ${IMG_SIZE} -gt ${MAX_IMG_SIZE} ]; then
```

Comment:
This needs to change to `MAX_IMG_SIZE_MB` now (and below).

Reply:
Oh. Yes.